geode-assembly/src/test/java/org/apache/geode/management/internal/cli/commands/StartServerCommandTest.java

```diff
@@ -141,4 +147,97 @@ public void testCreateServerCommandLineWithRestAPI() throws Exception {
+            .concat(File.pathSeparator).concat(StartMemberUtils.CORE_DEPENDENCIES_JAR_PATHNAME));
+    expectedCommandLineElements
+        .add("-DgemfirePropertyFile=".concat(gemfirePropertiesFile.getAbsolutePath()));
+    expectedCommandLineElements.add(
+        "-DgemfireSecurityPropertyFile=".concat(gemfireSecurityPropertiesFile.getAbsolutePath()));
+    expectedCommandLineElements.add("-Dgemfire.statistic-sample-rate=1500");
+    expectedCommandLineElements.add("-Dgemfire.disable-auto-reconnect=true");
+    expectedCommandLineElements.addAll(Arrays.asList(jvmArguments));
+    expectedCommandLineElements.add("-XX:OnOutOfMemoryError=kill -KILL %p");
+    expectedCommandLineElements.add("-Xms".concat(heapSize));
+    expectedCommandLineElements.add("-Xmx".concat(heapSize));
+    expectedCommandLineElements.add("-XX:+UseConcMarkSweepGC");
+    expectedCommandLineElements.add("-XX:CMSInitiatingOccupancyFraction="
+        .concat(String.valueOf(StartMemberUtils.CMS_INITIAL_OCCUPANCY_FRACTION)));
+    expectedCommandLineElements.add("-Dgemfire.launcher.registerSignalHandlers=true");
+    expectedCommandLineElements.add("-Djava.awt.headless=true");
+    expectedCommandLineElements.add("-Dsun.rmi.dgc.server.gcInterval=9223372036854775806");
+    expectedCommandLineElements.add("-Dgemfire.OSProcess.DISABLE_REDIRECTION_CONFIGURATION=true");
+    expectedCommandLineElements.add("org.apache.geode.distributed.ServerLauncher");
+    expectedCommandLineElements.add("start");
+    expectedCommandLineElements.add("fullServer");
+    expectedCommandLineElements.add("--assign-buckets");
+    expectedCommandLineElements.add("--debug");
+    expectedCommandLineElements.add("--disable-default-server");
+    expectedCommandLineElements.add("--force");
+    expectedCommandLineElements.add("--rebalance");
+    expectedCommandLineElements.add("--redirect-output");
+    expectedCommandLineElements.add("--server-port=41214");
+    expectedCommandLineElements.add("--spring-xml-location=/config/spring-server.xml");
+    expectedCommandLineElements.add("--critical-heap-percentage=95.5");
+    expectedCommandLineElements.add("--eviction-heap-percentage=85.0");
+    expectedCommandLineElements.add("--critical-off-heap-percentage=95.5");
+    expectedCommandLineElements.add("--eviction-off-heap-percentage=85.0");
+    expectedCommandLineElements.add("--max-connections=800");
+    expectedCommandLineElements.add("--max-message-count=500");
+    expectedCommandLineElements.add("--max-threads=100");
+    expectedCommandLineElements.add("--message-time-to-live=93");
+    expectedCommandLineElements.add("--socket-buffer-size=1048576");
+    expectedCommandLineElements.add("--hostname-for-clients=localhost");
+
+    assertNotNull(commandLineElements);
+    assertTrue(commandLineElements.length > 0);
+    assertEquals(commandLineElements.length, expectedCommandLineElements.size());
```

Comment:
Delete this assertion `expectedCommandLineElements` only contains a subset of `commandLineElements`.

Reply:
@pdxrunner: this is not accurate, `expectedCommandLineElements` contains **the full set of expected elements** in `commandLineElements`:

```
    Set<String> expectedCommandLineElements = new HashSet<>();
    expectedCommandLineElements.add(StartMemberUtils.getJavaPath());
    expectedCommandLineElements.add("-server");
    expectedCommandLineElements.add("-classpath");
    expectedCommandLineElements
        .add(StartMemberUtils.getGemFireJarPath().concat(File.pathSeparator).concat(customClasspath)
            .concat(File.pathSeparator).concat(StartMemberUtils.CORE_DEPENDENCIES_JAR_PATHNAME));
    expectedCommandLineElements
        .add("-DgemfirePropertyFile=".concat(gemfirePropertiesFile.getAbsolutePath()));
    expectedCommandLineElements.add(
        "-DgemfireSecurityPropertyFile=".concat(gemfireSecurityPropertiesFile.getAbsolutePath()));
    expectedCommandLineElements.add("-Dgemfire.statistic-sample-rate=1500");
    expectedCommandLineElements.add("-Dgemfire.disable-auto-reconnect=true");
    expectedCommandLineElements.addAll(Arrays.asList(jvmArguments));
    expectedCommandLineElements.add("-XX:OnOutOfMemoryError=kill -KILL %p");
    expectedCommandLineElements.add("-Xms".concat(heapSize));
    expectedCommandLineElements.add("-Xmx".concat(heapSize));
    expectedCommandLineElements.add("-XX:+UseConcMarkSweepGC");
    expectedCommandLineElements.add("-XX:CMSInitiatingOccupancyFraction="
        .concat(String.valueOf(StartMemberUtils.CMS_INITIAL_OCCUPANCY_FRACTION)));
    expectedCommandLineElements.add("-Dgemfire.launcher.registerSignalHandlers=true");
    expectedCommandLineElements.add("-Djava.awt.headless=true");
    expectedCommandLineElements.add("-Dsun.rmi.dgc.server.gcInterval=9223372036854775806");
    expectedCommandLineElements.add("-Dgemfire.OSProcess.DISABLE_REDIRECTION_CONFIGURATION=true");
    expectedCommandLineElements.add("org.apache.geode.distributed.ServerLauncher");
    expectedCommandLineElements.add("start");
    expectedCommandLineElements.add("fullServer");
    expectedCommandLineElements.add("--assign-buckets");
    expectedCommandLineElements.add("--debug");
    expectedCommandLineElements.add("--disable-default-server");
    expectedCommandLineElements.add("--force");
    expectedCommandLineElements.add("--rebalance");
    expectedCommandLineElements.add("--redirect-output");
    expectedCommandLineElements.add("--server-port=41214");
    expectedCommandLineElements.add("--spring-xml-location=/config/spring-server.xml");
    expectedCommandLineElements.add("--critical-heap-percentage=95.5");
    expectedCommandLineElements.add("--eviction-heap-percentage=85.0");
    expectedCommandLineElements.add("--critical-off-heap-percentage=95.5");
    expectedCommandLineElements.add("--eviction-off-heap-percentage=85.0");
    expectedCommandLineElements.add("--max-connections=800");
    expectedCommandLineElements.add("--max-message-count=500");
    expectedCommandLineElements.add("--max-threads=100");
    expectedCommandLineElements.add("--message-time-to-live=93");
    expectedCommandLineElements.add("--socket-buffer-size=1048576");
    expectedCommandLineElements.add("--hostname-for-clients=localhost");
```